src/app/leaderboard/ClientSideTable.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { FaCopy } from 'react-icons/fa';
+import { Data, Reward } from './types';
+
+interface ClientSideTableProps {
+  data: Data;
+}
+
+function shortenAddress(address?: string): string {
+  if (!address || address.length <= 10) return address || '';
+  return `${address.slice(0, 5)}...${address.slice(-4)}`;
+}
+
+function copyToClipboard(text: string) {
+  if (navigator.clipboard && navigator.clipboard.writeText) {
+    navigator.clipboard.writeText(text).catch((err) => alert(`Failed to copy text: ${err}`));
+  } else {
+    const textArea = document.createElement("textarea");
+    textArea.value = text;
+    textArea.style.position = 'fixed';
+    textArea.style.top = '-9999px';
+    textArea.style.left = '-9999px';
+    document.body.appendChild(textArea);
+    textArea.select();
+    try {
+      document.execCommand('copy');
+    } catch (err) {
+      alert(`Failed to copy text: ${err}`);
+    }
+    document.body.removeChild(textArea);
+  }
+}
+
+export default function ClientSideTable({ data }: ClientSideTableProps) {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isNarrowScreen, setIsNarrowScreen] = useState<boolean>(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsNarrowScreen(window.innerWidth < 1000);
+    };
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  const filteredResults = searchTerm
+    ? data.leaderboard_data.rewards.filter((reward) =>
+        reward.address.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    : data.leaderboard_data.rewards;
+
+  return (
+    <div className="max-w-screen-xl mx-3 md:mx-8">
+      {data.leaderboard_data?.block && (
+        <div className="ml-auto mr-auto text-zinc-400 mb-10">
+          Leaderboard synced to block {data.leaderboard_data.block}
+        </div>
+      )}
```

Comment:
Might as well pull in latest block from mempool as comparison as user has no context of current bitcoin block off top of their head